src/Polyline_TEST.cc

```diff
@@ -0,0 +1,176 @@
+  EXPECT_DOUBLE_EQ(0.2, polyline2.Height());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPolyline, CopyAssignmentAfterMove)
+{
+  sdf::Polyline polyline1;
+  polyline1.SetHeight(0.1);
+
+  sdf::Polyline polyline2;
+  polyline2.SetHeight(0.2);
+
+  // This is similar to what std::swap does except it uses std::move for each
+  // assignment
+  sdf::Polyline tmp = std::move(polyline1);
+  polyline1 = polyline2;
+  polyline2 = tmp;
+
+  EXPECT_DOUBLE_EQ(0.2, polyline1.Height());
+  EXPECT_DOUBLE_EQ(0.1, polyline2.Height());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPolyline, Load)
+{
+  sdf::Polyline polyline;
+  sdf::Errors errors;
+
+  // Null element
+  errors = polyline.Load(nullptr);
+  ASSERT_EQ(1u, errors.size());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_MISSING, errors[0].Code());
+  EXPECT_EQ(nullptr, polyline.Element());
+
+  // Bad element name
+  sdf::ElementPtr sdf(new sdf::Element());
+  sdf->SetName("banana");
+  errors = polyline.Load(sdf);
+  ASSERT_EQ(1u, errors.size());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_INCORRECT_TYPE, errors[0].Code());
+  EXPECT_NE(nullptr, polyline.Element());
+
+  // Missing <height> element
+  sdf->SetName("polyline");
+  errors = polyline.Load(sdf);
+  ASSERT_EQ(1u, errors.size());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_MISSING, errors[0].Code());
+  EXPECT_NE(std::string::npos, errors[0].Message().find("missing a <height>"));
+  EXPECT_NE(nullptr, polyline.Element());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPolyline, Points)
+{
+  sdf::Polyline polyline;
+  EXPECT_EQ(0u, polyline.PointCount());
+
+  ignition::math::Vector2d p1{1, 2};
+  ignition::math::Vector2d p2{3, 4};
+  ignition::math::Vector2d p3{5, 6};
+  ignition::math::Vector2d p4{7, 8};
+
+  EXPECT_TRUE(polyline.AddPoint(p1));
+  EXPECT_TRUE(polyline.AddPoint(p2));
+  EXPECT_TRUE(polyline.AddPoint(p3));
+
+  EXPECT_EQ(3u, polyline.PointCount());
+  EXPECT_EQ(3u, polyline.Points().size());
+  EXPECT_EQ(p1, *polyline.PointByIndex(0));
+  EXPECT_EQ(p2, *polyline.PointByIndex(1));
+  EXPECT_EQ(p3, *polyline.PointByIndex(2));
+  EXPECT_EQ(p1, polyline.Points()[0]);
+  EXPECT_EQ(p2, polyline.Points()[1]);
+  EXPECT_EQ(p3, polyline.Points()[2]);
+  EXPECT_EQ(nullptr, polyline.PointByIndex(3));
+
+  *polyline.PointByIndex(0) = p4;
+  EXPECT_EQ(p4, *polyline.PointByIndex(0));
+
+  polyline.ClearPoints();
+  EXPECT_EQ(0u, polyline.PointCount());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPolyline, ToElement)
+{
+  sdf::Polyline polyline;
+  polyline.SetHeight(1.2);
+
+  ignition::math::Vector2d p1{1, 2};
+  ignition::math::Vector2d p2{3, 4};
+  EXPECT_TRUE(polyline.AddPoint(p1));
+  EXPECT_TRUE(polyline.AddPoint(p2));
+
+  auto elem = polyline.ToElement();
+  ASSERT_NE(nullptr, elem);
+
+  sdf::Polyline polyline2;
+  polyline2.Load(elem);
+
+  EXPECT_DOUBLE_EQ(polyline.Height(), polyline2.Height());
+  ASSERT_DOUBLE_EQ(polyline.PointCount(), polyline2.PointCount());
```

Comment:
these are integer types, not double, so I think we can use `ASSERT_EQ` instead of `ASSERT_DOUBLE_EQ`

this may fix the windows warning

* https://build.osrfoundation.org/job/sdformat-ci-pr_any-windows7-amd64/3858/msbuild/

Reply:
https://github.com/ignitionrobotics/sdformat/pull/1003/commits/ca97fa9a8e244a61f78170e295c9ab40269d2b7d